utils/tzif/src/lib.rs

```diff
@@ -14,35 +14,53 @@
 //!
 //! ### Parse TZif Files
 //! ```no_run
-//! use combine::{Parser, stream};
-//! use std::fs::File;
-//! use tzif::tzif;
-//!
-//! let file = File::open("path_to_file").unwrap();
-//! let stream = stream::buffered::Stream::new(
-//!     stream::position::Stream::new(stream::read::Stream::new(file)),
-//!     0, /* lookahead */
-//! );
-//! let data = tzif().parse(stream).unwrap();
+//! use tzif::TzifParser;
+//! let data = TzifParser::parse_file("path_to_file").unwrap();
 //! ```
 //!
 //! ### Parse POSIX time-zone strings
 //! ```rust
-//! use combine::Parser;
-//! use tzif::posix_tz_string;
-//!
-//! let data = posix_tz_string()
-//!     .parse(b"WGT3WGST,M3.5.0/-2,M10.5.0/-1".as_slice())
-//!     .unwrap();
+//! use tzif::PosixParser;
+//! let data = PosixParser::parse_bytes(b"WGT3WGST,M3.5.0/-2,M10.5.0/-1").unwrap();
 //! ```
 
 #![warn(missing_docs)]
 
+use combine::{stream, Parser};
+use data::{posix::PosixTzString, tzif::TzifData};
+use error::Error;
+use std::fs::File;
+use std::path::Path;
+
 /// The parsed data representations.
 pub mod data;
 
 /// The parser implementations.
 pub mod parse;
 
-pub use parse::posix::posix_tz_string;
-pub use parse::tzif::tzif;
+/// Error types an implementations.
+pub mod error;
+
+/// A parser for [Time Zone Information Format (`TZif`)](https://tools.ietf.org/id/draft-murchison-tzdist-tzif-00.html) files.
+pub struct TzifParser {}
+
+impl TzifParser {
+    /// Parses a `TZif` file at the provided `path`.
+    pub fn parse_file<P: AsRef<Path>>(path: P) -> Result<TzifData, Error> {
```

Comment:
thought: Should these just be toplevel public functions (`parse_tzif_file` and `parse_posix_file`)?

Typically when I see a `Parser` object I imagine a mutable object that holds state

current design is mostly fine though

Reply:
Yeah, that's reasonable. 

I'll make the changes.